Frontend rewrite to Vue

KFBI1706 opened the pull request:
Rewrote the frontend JS to Vue.JS. Mainly because we needed a search function, but I ended up rewriting the rest aswell. 

As for features, it's on par with the previous frontend with these changes:
- Search, you can now search using the name, organization or the URL
- Errors, errors are now setup as a list to accomodate the various problems that could be going on
- Progress animation with status
- Form validation and gradual apperance of the required elements of the web page
- Styling

I'm not a frontend dev by any stretch of the imagination, so critique is very welcome!

KFBI1706 commented:
Some screenshots from the new page
![Screenshot 2019-07-03 at 09 21 45](https://user-images.githubusercontent.com/4007953/60572075-1aed3b00-9d75-11e9-965c-bd37af8044d7.png)
![Screenshot 2019-07-03 at 09 21 31](https://user-images.githubusercontent.com/4007953/60572080-1d4f9500-9d75-11e9-80a0-256fe6552382.png)

ambientsound commented on hookd/pkg/auth/submit.go:
```diff
@@ -49,10 +47,19 @@ func (h *SubmittedFormHandler) ServeHTTP(w http.ResponseWriter, r *http.Request)
+	fullName := r.Form.Get("repository")
 	teamNames := r.Form["team[]"]
 
+	arr := strings.Split(fullName, "/")
```
This is already implemented in `github.SplitFullname`.

ambientsound commented on hookd/templates/form.html:
```diff
@@ -1,186 +1,162 @@
+                <b>Please correct the following error(s):</b>
+                <ul class="errors">
+                    <li class="error" v-for="error in errors">[[ error ]]</li>
+                    </ulc>
```
Typo

ambientsound commented on hookd/templates/site.html:
```diff
@@ -7,6 +7,8 @@
     <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500">
     <link rel="stylesheet" href="/assets/css/style.css">
+    <script src="/assets/js/vue.js"></script>
+    <script src="/assets/js/axios.min.js"></script>
```
Could we extract the Javascript code and put it as a separate file?

KFBI1706 commented:
* The content should be centered on page, but not `text-align: center`
I've aligned a lot of the text to the left now, but still kept some text center alligned as I thought it looked better.
* There is too much jumping around when clicking UI elements. Stuff should stay in the same position.
I've tried to reduce this aswell. And the sporadic jumping when the status was showing up briefly is fixed and I think it looks decent now. If we want the UI elements to be even more static, we'd have to preallocated containers for the different elements, but I personlally don't think that looks good. 
* The search box should have a border to make it look like a search box. It should also be positioned directly below the logo.
Fixed.
* The search box should be automatically focused when loading the page.
Fixed.
* We are still lacking some documentation.
I'm unsure here; should we create a doc.html template and link to it in a navbar at the top of the page?
What should the contents of the doc be?
* Sorting should be case insensitive.
Did the insensitive sort in the go code before it's passed to the frontend.
* The `clear` button isn't really useful.
Removed.
* Tests?
Yes. Working on this next

KFBI1706 commented:
**Features**
* increase width and align elements on the left
* Several computed props
* Added a more results button incase the user get's a lot of results and isn't quite sure of the name
* move errorlist to bottom
* added no result text
* Fully clickable radio-buttons and checkboxes
* Better transition between loading, errors and submissions, reduce the amount of jumping around on the screen

**Bug Fixes**
* Reset teams between repository selection

The last point of contention is if the repository name should be filled into the searchbar once clicked. For now I'm keeping it that way, both because I have a bunch of logic relying on it. Aswell as it looks a lot better with the one repo selected and the available teams / the error page just beneath that.

If we go for the other solution The team and submit button wil most likely go past the screen and you'd have to scroll and everything would just look more cluttered.
